### PR TITLE
Do not pushdown predicate when it contains CTEAnchor

### PR DESCRIPTION
**Pull request by asubramanya**
If there is a predicate which contains a CTEAnchor, pushing it down can cause
duplicate CTEProducers to be generated in the plan. So for now disable pushing
down the predicate in such cases.

### CODE REVIEW

**Comment by asubramanya** [comment]
@hardikar It seems to work fine for the query you have posted. I don't see any duplicate CTE producers in the plan.

File: libgpopt/src/operators/CNormalizer.cpp [hunk 1]
```diff
@@ -1151,7 +1151,7 @@ CNormalizer::PushThru
 	{
 		CExpression *pexprConj = (*pdrgpexprConjuncts)[ul];
 		pexprConj->AddRef();
-		if (FPushable(pexprLogical, pexprConj))
+		if (FPushable(pexprLogical, pexprConj) && !CUtils::FHasCTEAnchor(pexprConj))
```
Comment (chrishajas) [inline libgpopt/src/operators/CNormalizer.cpp]: Might be good to add a comment saying why we can't push down CTE anchors (or maybe even put this function call inside `FPushable()`)

Comment (asubramanya) [reply libgpopt/src/operators/CNormalizer.cpp]: Done.

File: libgpopt/src/base/CUtils.cpp [hunk 2]
```diff
@@ -897,6 +897,32 @@ CUtils::FHasSubquery
 	return pexpr->DeriveHasSubquery();
 }
 
+// check for existence of CTE anchor
+BOOL
+CUtils::FHasCTEAnchor
+(
+ CExpression *pexpr
+ )
```
[the remark this answers is not in the document]
Comment (asubramanya) [reply libgpopt/src/base/CUtils.cpp]: Done

**Pull request merged**